src/Commands/ModelChatCommand.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\llm_services\Commands;
+
+use Drupal\llm_services\Model\Message;
+use Drupal\llm_services\Model\MessageRoles;
+use Drupal\llm_services\Model\Payload;
+use Drupal\llm_services\Plugin\LLModelProviderManager;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Question\Question;
+
+/**
+ * Chat with a model through a provider.
+ */
+class ModelChatCommand extends Command {
+
+  /**
+   * Default constructor.
+   *
+   * @param \Drupal\llm_services\Plugin\LLModelProviderManager $providerManager
+   *   The provider manager.
+   */
+  public function __construct(
+    private readonly LLModelProviderManager $providerManager,
+  ) {
+    parent::__construct();
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  protected function configure(): void {
+    $this
+      ->setName('llm:model:chat')
+      ->setDescription('Chat with model (use ctrl+c to stop chatting)')
+      ->addUsage('llm:model:chat ollama llama3')
+      ->addArgument(
+        name: 'provider',
+        mode: InputArgument::REQUIRED,
+        description: 'Name of the provider (plugin).'
+      )
+      ->addArgument(
+        name: 'name',
+        mode: InputArgument::REQUIRED,
+        description: 'Name of the model to use.'
+      )
+      ->addOption(
+        name: 'system-prompt',
+        mode: InputOption::VALUE_REQUIRED,
+        description: 'System message to instruct the llm have to behave.',
+        default: 'Use the following pieces of context to answer the users question. If you don\'t know the answer, just say that you don\'t know, don\'t try to make up an answer.'
+      )
+      ->addOption(
+        name: 'temperature',
+        mode: InputOption::VALUE_REQUIRED,
+        description: 'The temperature of the model. Increasing the temperature will make the model answer more creatively.',
+        default: '0.8'
+      )
+      ->addOption(
+        name: 'top-k',
+        mode: InputOption::VALUE_REQUIRED,
+        description: 'Reduces the probability of generating nonsense. A higher value (e.g. 100) will give more diverse answers.',
```

Comment:
I read the first sentence as if a higher value reduce nonsense, but the second sentence reveals the opposite.

Reply:
yes, it is taken directly from Ollama documentation...